src/XRP/wallet-factory.ts

```diff
@@ -25,10 +31,39 @@ export default class WalletFactory {
     this.isTest = XrpUtils.isTestNetwork(network)
   }
 
+  /**
+   * Generate a new hierarchical deterministic wallet from a seed and derivation path.
+   *
+   * @param seed - A hex encoded seed string.
+   * @param derivationPath - The given derivation path to use. If undefined, the default path is used.
+   * @returns A new wallet from the given mnemonic if the mnemonic was valid, otherwise undefined.
+   */
+  public walletFromSeedAndDerivationPath(
+    seed: string,
+    derivationPath = WalletFactory.defaultDerivationPath,
+  ): Wallet | undefined {
+    try {
+      const seedBytes = Utils.toBytes(seed)
+      const seedBuffer = Buffer.from(seedBytes)
+      const masterNode = bip32.fromSeed(seedBuffer)
+      const node = masterNode.derivePath(derivationPath)
+      if (node.privateKey === undefined) {
+        return undefined
+      }
+
+      const publicKey = Utils.hexFromBuffer(node.publicKey)
+      const privateKey = Utils.hexFromBuffer(node.privateKey)
+      return new Wallet(publicKey, `00${privateKey}`, this.isTest)
```

Comment:
Ultimately, we want to generate keys that can be used with `ripple-pairs`. Rippled generates private keys with a 1 byte prefix representing whether the key is secp256k1 or ed25519. The former uses bytes 0x00 and the latter uses 0xED. 

You can see these prefixes here:
https://github.com/ripple/ripple-keypairs/blob/7e67cbb9e1cc293bf696b0e0b054a214d607d28b/src/index.ts#L44

0xED is also mentioned here: https://xrpl.org/cryptographic-keys.html#ed25519-key-derivation. 

I append this prefix here because the generic libraries (bip32) don't have the concept of this seed. I could append in wallet internally, but then `walletFromSeed` which uses `ripple-keypairs` does give me this prefix this one-off logic would just move elsewhere. 

LMK what you think. After thinking about this, maybe it's "more correct" to give an unprefixed public key to the wallet and force an enum. That way when folks say "wallet.privateKey" they get a private key that works for all libraries (though not ripple-keypairs).

Reply:
> Rippled generates private keys with a 1 byte prefix representing whether the key is secp256k1 or ed25519. The former uses bytes 0x00 and the latter uses 0xED.

Helpful. I would at the least add that as a code comment above this line.

> Maybe it's "more correct" to give an unprefixed public key to the wallet and force an enum. That way when folks say "wallet.privateKey" they get a private key that works for all libraries.

I have no idea what usage looks like in the real world, but I would imagine that most people interacting with XRPL just use `ripple-lib` & co, right?